src/mlpack/core/optimizers/minibatch_sgd/minibatch_sgd.hpp

```diff
@@ -165,6 +166,9 @@ class MiniBatchSGDType
   //! are reset before Optimize call.
   bool& ResetPolicy() { return resetPolicy; }
 
+  //! Get the current batch being processed.
+  size_t& CurrentBatch() { return currentBatch; }
```

Comment:
Since the Optimizer API influence on the whole library, I think that any changes of the API require a separate discussion and consequently a separate PR. If I am not mistaken only `mini-batch SGD` supports mini-batches right now. So, I guess there is no sense to change the API until other optimizers allow to use mini-batches.